### PR TITLE
#228_Create_resolver_for_app_user

### PR DESCRIPTION
**Pull request by pergaliuke**
Not finished, wanted to check if solution would be good enough. As well I hoped to use inheritResolversFromInterfaces, but it doesn't work.

### CODE REVIEW

**Comment by pergaliuke** [comment]
Error output is not being shown in docker logs #240.
addResolversToSchema handles schema validation, errors are visible in docker logs. For example.
image

File: src/schema/index.ts [hunk 1]
```diff
@@ -52,8 +56,9 @@ const schema = loadSchemaSync('src/schema/typeDefs/*.graphql', {
         animalFoundEventResolver,
         givenAwayEventResolvers,
         favoriteAnimalResolvers,
+        givenAwayEventResolvers,
```
Comment (AzisK) [inline src/schema/index.ts:59]: I see `givenAwayEventResolvers` 2 times

File: src/schema/typeDefs/animalRegistration.graphql [hunk 2]
```diff
@@ -3,6 +3,10 @@ enum RegistrationStatus {
     Inactive
 }
 
+extend type Query {
```
Comment (AzisK) [inline src/schema/typeDefs/animalRegistration.graphql:6]: Is this part of this issue?

File: test/event.graphql.test.ts [hunk 3]
```diff
@@ -7,6 +7,16 @@ require('dotenv').config({ path: './test/.env' });
 const url = process.env.TEST_URL || 'http://localhost:8081';
 const request = supertest(url);
 
+// eslint-disable-next-line import/prefer-default-export
+export const authorFields = `
```
Comment (AzisK) [inline test/event.graphql.test.ts:11]: Maybe this is a good candidate to go into a separate file?

**Review by AzisK — CHANGES_REQUESTED** [review]
Maybe we should not return all the information but otherwise it looks good to me.

Left a few comments and questions

File: src/schema/typeDefs/animalRegistration.graphql (hunk 2 above)
Comment (pergaliuke) [reply src/schema/typeDefs/animalRegistration.graphql:6]: After I changed to addResolversToSchema, app crashed
image

I fixed it by adding to schema. But maybe we can remove it from resolvers and from schema if this functionality is not needed anymore.

File: src/schema/index.ts (hunk 1 above)
Comment (pergaliuke) [reply src/schema/index.ts:59]: Good catch :)

File: test/event.graphql.test.ts (hunk 3 above)
Comment (pergaliuke) [reply test/event.graphql.test.ts]: Done

File: src/index.ts [hunk 4]
```diff
@@ -82,8 +82,23 @@ initClients().then(({ pgClient, cloudinaryClient }) => {
...
         schema,
+        formatResponse: (response: any) => {
+            if (response.errors) {
+                console.log(response);
+                // eslint-disable-next-line default-case
+                switch (response.errors[0].extensions.code) {
+                case 'GRAPHQL_VALIDATION_FAILED':
+                case 'BAD_USER_INPUT':
+                    console.log(response?.http?.status);
+                    // response.http.status.set(400);
```
Comment (JustasBelevicius) [inline src/index.ts:93]: Well done! One note here: either remove the console statement, or remove the commented out line.

**Review by JustasBelevicius — APPROVED** [review]

**Review by AzisK — APPROVED** [review]

**Pull request merged**